ckan/lib/i18n.py

```diff
@@ -131,6 +138,27 @@ def handle_request(request, tmpl_context):
         config.get('ckan.locale_default', 'en')
     if lang != 'en':
         set_lang(lang)
+
+    for plugin in PluginImplementations(ITranslation):
+        if lang in plugin.locales():
+            translator = Translations.load(
+                dirname=plugin.directory(),
+                locales=lang,
+                domain=plugin.domain()
+            )
+            try:
+                pylons.translator.merge(translator)
+            except AttributeError:
+                # this occurs when an extension has 'en' translations that
+                # replace the default strings. As set_lang has not been run,
+                # pylons.translation is the NullTranslation, so we have to
+                # replace the StackedObjectProxy ourselves manually.
+                environ = pylons.request.environ
+                environ['pylons.pylons'].translator = translator
+                if 'paste.registry' in environ:
+                    environ['paste.registry'].replace(pylons.translator,
+                                                      translator)
```

Comment:
the other option to this would be to remove all the `if lang != 'en'` statements, this would probably also mean maintaining an empty i18n/en/LC_MESSAGES/ckan.mo/po file